Automattic/woocommerce-payments
# Fix translations in transaction/deposit exports

Fixes #1976 

#### Changes proposed in this Pull Request
As noted in #1976 Transaction type column fails to translate to the respective language when exporting transactions to a CSV, this also happens for Type and Status fields when exporting Deposits. 
[csv-export](https://github.com/woocommerce/woocommerce-admin/tree/main/packages/csv-export) library expects a table row object as input and relies on the `value` property to populate export content. Changes proposed maps the table row `value` using `display*` mappers, as we are already doing [here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/transactions/list/index.tsx#L319)

Another approach would be to populate a separate object specifically to be used in `generateCSVDataFromTable` instead of reusing `rows` so that we can prevent mutating the `value`. I opted to go with the proposed approach as we are not using `rows` anywhere else and we are already mutating `value` for certain columns. But would be happy to switch to the second approach or try out another.










#### Testing instructions





1. In a site with existing WC Pay transactions, change site language to one German, Italian, Spanish, French from Settings -> General -> Site Language
2. Export transactions from Payments -> Transactions -> Download
3. Ensure that Type column is correctly translated and matches what's shown on the browser
4. Export deposits from Payments -> Deposists -> Download
5. See that Type and Status fields are translated and matches what's shown on the browser.
-------------------


**Post merge**
